src/Accelerators/NNPA/Conversion/ZLowToLLVM/ZLowToLLVMCommon.cpp

```diff
@@ -90,6 +91,71 @@ ApiRegistry RegisterAllApis(MLIRContext *context) {
   return registry;
 }
 
+#define CASE_FOR_APIIDSTR(apiId)                                               \
+  case API::apiId:                                                             \
+    return #apiId
+std::string apiIdStr(API apiId) {
+  switch (apiId) {
+    CASE_FOR_APIIDSTR(NULL_API);
+    CASE_FOR_APIIDSTR(ZDNN_INIT_PRE_TRANSFORMED_DESC);
+    CASE_FOR_APIIDSTR(ZDNN_GENERATE_TRANSFORMED_DESC);
+    CASE_FOR_APIIDSTR(ZDNN_GENERATE_TRANSFORMED_DESC_CONCATENATED);
+    CASE_FOR_APIIDSTR(ZDNN_GETSIZE_ZTENSOR);
+    CASE_FOR_APIIDSTR(ZDNN_TRANSFORM_ZTENSOR);
+    CASE_FOR_APIIDSTR(ZDNN_TRANSFORM_ORIGTENSOR);
+    CASE_FOR_APIIDSTR(ZDNN_ADD);
+    CASE_FOR_APIIDSTR(ZDNN_SUB);
+    CASE_FOR_APIIDSTR(ZDNN_MUL);
+    CASE_FOR_APIIDSTR(ZDNN_DIV);
+    CASE_FOR_APIIDSTR(ZDNN_MIN);
+    CASE_FOR_APIIDSTR(ZDNN_MAX);
+    CASE_FOR_APIIDSTR(ZDNN_LOG);
+    CASE_FOR_APIIDSTR(ZDNN_EXP);
+    CASE_FOR_APIIDSTR(ZDNN_RELU);
+    CASE_FOR_APIIDSTR(ZDNN_TANH);
+    CASE_FOR_APIIDSTR(ZDNN_SIGMOID);
+    CASE_FOR_APIIDSTR(ZDNN_SOFTMAX);
+    CASE_FOR_APIIDSTR(ZDNN_LSTM);
+    CASE_FOR_APIIDSTR(ZDNN_GRU);
+    CASE_FOR_APIIDSTR(ZDNN_MATMUL_OP);
+    CASE_FOR_APIIDSTR(ZDNN_MATMUL_BCAST_OP);
+    CASE_FOR_APIIDSTR(ZDNN_CONV2D);
+    CASE_FOR_APIIDSTR(ZDNN_AVGPOOL2D);
+    CASE_FOR_APIIDSTR(ZDNN_MAXPOOL2D);
+    CASE_FOR_APIIDSTR(ZDNN_MEANREDUCE2D);
+    CASE_FOR_APIIDSTR(ZDNN_BATCHNORM);
+  default:
+    break;
+  }
+  return "INVALID_APIID";
+}
+
+// Emit code for `IF lhs != rhs THEN print messages and exit`
+void equalOrExit(ModuleOp &module, PatternRewriter &rewriter, Location loc,
+    Value lhs, Value rhs, std::string errorMsg = "",
+    bool funcCallErrorExit = false) {
+  MultiDialectBuilder<LLVMBuilder> create(rewriter, loc);
+  create.llvm.ifThenElse(/*cond=*/
+      [&](LLVMBuilder &createLLVM) {
+        return createLLVM.icmp(LLVM::ICmpPredicate::ne, lhs, rhs);
+      }, /*then=*/
+      [&](LLVMBuilder &createLLVM) {
+        MultiDialectBuilder<LLVMBuilder, KrnlBuilder, MathBuilder> create(
+            createLLVM);
+        // Print an error message.
+        create.krnl.printf(StringRef(errorMsg), rhs, StringRef("%#x"), true);
+
+        // Exit
+        if (funcCallErrorExit) {
+          MLIRContext *context = rewriter.getContext();
+          Type int32Ty = IntegerType::get(context, 32);
```

Comment:
Please use i64 since we have an issue with i32 on big endian machine. More info: https://github.com/onnx/onnx-mlir/issues/778